### PR TITLE
Fix cancellation of mandatory install

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

- Create installer abstraction
- Fallback to old method (MIUI optimizations workaround)
- Fix tracking cancel installing

## Related PRs or issues

#1619 / [AB#92886](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/92886)
https://github.com/microsoft/appcenter-sdk-dotnet/issues/1633 / [AB#91794](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/91794)

### CODE REVIEW

File: sdk/build.gradle [hunk 1]
```diff
@@ -63,7 +63,7 @@ subprojects {
         testOptions {
             unitTests {
                 all {
-                    jvmArgs '-noverify', '-Djdk.attach.allowAttachSelf=true'
+                    jvmArgs '-noverify', '-Djdk.attach.allowAttachSelf=true', '-Djdk.module.illegalAccess.silent=true'
```
Comment (DmitriyKirakosyan) [inline sdk/build.gradle:66]: do we need this flag?

Comment (MatkovIvan) [reply sdk/build.gradle:66]: Yes, it ignores warning that was added in new Java about black magic in PowerMock

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 2]
```diff
@@ -763,213 +707,207 @@ private synchronized void cancelPreviousTasks() {
...
-            if (mInstallInProgress) {
-                AppCenterLog.info(LOG_TAG, "Installing in progress...");
-                return;
-            }
+        /* The installer takes care of resume event during installing a new release. */
+        if (mReleaseInstaller != null) {
+            mReleaseInstaller.resume();
+            return;
+        }
+        AppCenterLog.debug(LOG_TAG, "Resume distribute workflow...");
```
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:735]: It's intended to be here, but it's not a big deal

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 3]
```diff
@@ -1983,19 +1849,36 @@ synchronized void setDownloading(@NonNull ReleaseDetails releaseDetails, long en
...
-            SharedPreferencesManager.putInt(PREFERENCE_KEY_DOWNLOAD_STATE, DOWNLOAD_STATE_INSTALLING);
-        } else {
-            completeWorkflow(releaseDetails);
+
+        /*
+         * If we already notified, that means this check was triggered by application being resumed,
+         * thus in foreground at the moment the check download async task was started.
+         *
+         * We should not hold the install any longer now, even if the async task was long enough
+         * for app to be in background again, we should show install U.I. now.
```
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:1862]: The comment isn't new + there are a lot of similar places. Keeping it as is

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**